src/bin/idl/mod.rs

```diff
@@ -281,9 +282,13 @@ fn instruction(
             .unwrap()
             .1;
 
+        // The anchor discriminator is what Solidity calls a selector
+        let selector = discriminator(if state { "state" } else { "global" }, &instr.name);
+
         write!(
             f,
-            "\tfunction {}(",
+            "\t@selector([{}])\n\tfunction {}(",
+            selector.iter().map(|v| format!("{:#02x}", v)).join(","),
```

Comment:
We should have a test like `codegen` for this file, i.e., you generate the interface from an IDL and check if it is correct.

Reply:
We have the the anchor integration test - if this doesn't work, then the test will fail.